packages/@aws-cdk/aws-config/lib/rule.ts

```diff
@@ -383,3 +391,356 @@ export class CustomRule extends RuleNew {
     }
   }
 }
+
+
+/**
+ * Resources types that are supported by AWS Config
+ * @see https://docs.aws.amazon.com/config/latest/developerguide/resource-config-reference.html
+ */
+export class ResourceType {
+  /** APIGGateway Stage which is contained in ApiGateway Rest Api or associated with WAFRegional WebACL */
+  public static readonly APIGATEWAY_STAGE = new ResourceType('AWS::ApiGateway::Stage');
+
+  /** APIGatewayv2 stage contained in ApiGatewayV2 Api */
+  public static readonly APIGATEWAYV2_STAGE = new ResourceType('AWS::ApiGatewayV2::Stage');
+
+  /** APIGateway REST Api that contains an ApiGateway Stage */
+  public static readonly APIGATEWAY_REST_API = new ResourceType('AWS::ApiGateway::RestApi');
+
+  /** API Gatewayv2 API that contains an ApiGatewayV2 Stage */
+  public static readonly APIGATEWAYV2_API = new ResourceType('AWS::ApiGatewayV2::Api');
+
+  /** CloudFront distribution which is associated with an Amazon CloudFront distribution */
+  public static readonly CLOUDFRONT_DISTRIBUTION = new ResourceType('AWS::CloudFront::Distribution');
+
+  /** CloudFront streaming distribution which is associated with an Amazon CloudFront streaming distribution */
+  public static readonly CLOUDFRONT_STREAMING_DISTRIBUTION = new ResourceType('AWS::CloudFront::StreamingDistribution');
+
+  /** Amazon CloudWatch Alarm */
+  public static readonly CLOUDWATCH_ALARM = new ResourceType('AWS::CloudWatch::Alarm');
+
+  /** Amazon DynamoDB Table */
+  public static readonly DYNAMODB_TABLE = new ResourceType('AWS::DynamoDB::Table');
+
+  /** Elastic Block Store (EBS) volume which is attached to an EC2 instance */
+  public static readonly EBS_VOLUME = new ResourceType('AWS::EC2::Volume');
+
+  /** EC2 host which contains an EC2 instance */
+  public static readonly EC2_HOST = new ResourceType('AWS::EC2::Host');
+
+  /** EC2 Elastic IP attached to an EC2 instance */
+  public static readonly EC2_EIP = new ResourceType('AWS::EC2::EIP');
+
+  /**
+   * EC2 instance which:
+   *  contains an EC2 network interface,
+   *  is associated with an EC2 security group,
+   *  is attached to an Amazon EBS volume or EC2 Elastic IP EIP),
+   *  is contained in an EC2 Dedicated host, route table, subnet, VPC
+   */
```

Comment:
This looks like you're trying to capture the tables present in the [resource types](https://docs.aws.amazon.com/config/latest/developerguide/resource-config-reference.html) documentation here in docstring format.

I think this format is actually a bit more confusing than helpful. Do I need an ENI or EBS volume for an instance to work properly with Config? This phrasing mildly implies it. I'd keep it simple here, and link to the docs for more context. For example, just keep these docstrings as '${Service Name} ${Resource Type}':

```suggestion
  /** EC2 instance */
```

Reply:
will simplify to only go with resource name